### ISSUE TITLE
Postgres JSON operator is sometimes unparsable

### ISSUE THREAD

**Issue by douglascook** [sqlfluff/sqlfluff#5128]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Using the postgres dialect: selecting a field from a JSON column, where the table is referenced and there is no whitespace around the operator results in an `unparsable` error:

```sql
-- Unparsable
SELECT a_table.a_column->'field'
FROM my_schema.a_table;

-- Works without a reference to the table
SELECT a_column->'field'
FROM my_schema.a_table;

-- Works with table qualifier as long as there are spaces around the operator
SELECT a_table.a_column -> 'field'
FROM my_schema.a_table;
```

### Expected Behaviour

Query should be parsed successfully

### Observed Behaviour

Query described above cannot be parsed when running, eg. `sqlfluff lint`

### How to reproduce

**json_operator_query.sql**:

```sql
SELECT a_table.a_column->'field'
FROM my_schema.a_table;
```

Run `sqlfluff lint --dialect postgres json_operator_query.sql`

### Dialect

Postgres

### Version

Broken since release `2.0.3`. I managed to narrow down the regression to 532a234. 

I would have liked to find a fix myself and open a PR, but the issue is in the parser so it looks like it will be less straightforward to fix without some understanding of how the parser works.

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ooobii** [comment]
This is also true for the `->>` operator. There is a workaround by casting to `json` first though!

By updating the json_operator_query.sql to look like this:
```sql
SELECT a_table.a_column::json->'field'
FROM my_schema.a_table;
```

and by running the SQLFluff command `sqlfluff lint --dialect postgres json_operator_query.sql`, it passes. This is also the case for `::json->>'field'` operator as well.

A cast using `::json` should **not** be required in the Postgres dialect for json property/accessor uses. 😄

**Comment by ooobii** [comment]
Also casting to `jsonb` works, too!

**Comment by douglascook** [comment]
I just tested this again with version `2.3.5` and it's now working as expected. Not sure which release it was fixed in but thank you to whoever fixed it! 🙌
